--- src/lread.c.orig	2018-06-06 23:37:40.763781400 +0800
+++ src/lread.c	2018-06-06 23:22:31.316272600 +0800
@@ -4619,7 +4619,12 @@
         } /* Vinstallation_directory != Vsource_directory */
 
     } /* if Vinstallation_directory */
-
+  else
+    {
+      Vsource_directory
+        = Fexpand_file_name (build_string ("../"),
+                             Fcar (decode_env_path (0, PATH_DATA, 0)));
+    }
   return lpath;
 }
 
